### PR TITLE
hotfix/gtms-upload-clinic-capacity-data-jsdoc

### PR DESCRIPTION
**Pull request by fais-2**
## Description
https://nhsd-jira.digital.nhs.uk/browse/GAL-401

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js [hunk 1]
```diff
@@ -78,7 +78,15 @@ export const handler = async (event, context) => {
...
 
-// METHODS
+/**
+ * This function is used to retrieve an object from S3,
+ * and allow the data retrieved to be used in your code.
+ * @function readCsvFromS3
+ * @param {string} bucketName The name of the bucket you are querying
+ * @param {string} key The name of the object you are retrieving
+ * @param {S3Client} client Instance of S3 client
+ * @returns {Object} The data of the file you retrieved
```
Comment (chenglawson1) [inline terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js:88]: Function returns a string, please change return type from Object to string.
Function can also throw an error, please add @throws.

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js [hunk 2]
```diff
@@ -93,7 +101,16 @@ export const readCsvFromS3 = async (bucketName, key, client) => {
...
-
+/**
+ * This function is used to write a new object in S3
+ * @function pushCsvToS3
+ * @param {string} bucketName The name of the bucket you are pushing to
+ * @param {string} body The data you will be writing to S3
+ * @param {string} key The name you want to give to the file you will write to S3
+ * @param {string} rejectedReason The rejected reason
+ * @param {S3Client} client Instance of S3 client
+ * @returns {Object} metadata about the request, including httpStatusCode
```
Comment (chenglawson1) [inline terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js:112]: Description needs to change to response.
Please add @throws.

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js [hunk 3]
```diff
@@ -118,6 +135,15 @@ export const pushCsvToS3 = async (
...
 };
 
+/**
+ * Queries DynamoDB table and returns items found based on primary key field/value
+ * @async
+ * @function getItemsFromTable
+ * @param {string} table Table name
+ * @param {DynamoDBClient} client Instance of DynamoDB client
+ * @param {string} key Primary key value
+ * @returns {Array} Items returned from query
```
Comment (chenglawson1) [inline terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js:145]: Type returned is not an Array, the array is in an Object.

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js [hunk 4]
```diff
@@ -143,7 +178,17 @@ const checkPhlebotomy = async (payload, arr) => {
...
+/**
+ * Saves an object to the Phlebotomy table.
+ *
+ * @async
+ * @function saveObjToPhlebotomyTable
+ * @param {Object} MeshObj - The mesh object to be saved to the Phlebotomy table.
+ * @param {string} environment - The environment name.
+ * @param {DynamoDBClient} client - The database client used to interact with the Phlebotomy table.
+ * @param {string} clinicName - The name of the clinic associated with the object.
+ * @returns {Array} updated Clinic with item to the Phlebotomy table.
```
Comment (chenglawson1) [inline terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js:190]: Return type is boolean not Array.

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js (hunk 3 above)
Comment (fais-2) [reply terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js]: @chenglawson1  resolved

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js (hunk 4 above)
Comment (fais-2) [reply terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js]: @chenglawson1  resolved

File: terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js (hunk 2 above)
Comment (fais-2) [reply terraform/src/gtmsUploadClinicCapacityDataLambda/gtmsUploadClinicCapacityDataLambda.js]: @chenglawson1  resolved

**Review by chenglawson1 — APPROVED** [review]

**Review by andrew-cleveland — APPROVED** [review]

**Pull request merged**